tools/dds/dds-adapter/rs-dds-adapter.cpp

```diff
@@ -151,24 +166,84 @@ try
         std::shared_ptr< dds_device_server > server;
         std::shared_ptr< tools::lrs_device_controller > controller;
     };
     std::map< rs2::device, device_handler > device_handlers_list;
-    
+
     std::cout << "Start listening to RS devices.." << std::endl;
 
     // Create a RealSense context
     settings["dds"] = false;                  // We want non-DDS (local) devices only
     settings["format-conversion"] = "basic";  // Conversion of raw sensor formats will be done by the receiver
     rs2::context ctx( settings.dump() );
 
+    // Set up the ros_discovery_info broadcaster
+    rsutils::concurrency::delayed broadcast_ros_discovery_info(
+        [&]
+        {
+            realdds::topics::ros2::participant_entities_info_msg msg;
+
+            // The RMW context gid is initialized, at least with the FastRTPS implementation, from the participant's GUID (see
+            // init_context_impl()), so the GID of the "ParticipantEntitiesInfo" is simply the participant GUID.
+            msg.set_gid( participant->guid() );
+
+            std::string const node_namespace = realdds::topics::ros2::NAMESPACE;
+            for( auto const & dev2handler : device_handlers_list )
+            {
+                auto & device = dev2handler.second;
+                if( device.info.serial_number().empty() )
+                    continue;  // E.g., recovery device
+                std::string node_name = build_node_name( device.info.name(), device.info.serial_number() );
+                topics::ros2::participant_entities_info_msg::node node( node_namespace, node_name );
+                device.controller->fill_ros2_node_entities( node );
+                msg.add( std::move( node ) );
+            }
+
+            msg.write_to( *discovery_info_writer );
+        } );
+
+    // And a writer for the ROS2 /parameter_events topics
+    std::shared_ptr< realdds::dds_topic_writer > parameter_events_writer;
+    std::string const
+        parameters_events_topic_name = rsutils::string::from()
+                                    << topics::ros2::ROOT
+                                    << topics::ros2::PARAMETER_EVENTS_NAME;
+    if( auto topic = realdds::topics::ros2::parameter_events_msg::create_topic(
+        participant,
+        parameters_events_topic_name.c_str() ) )
+    {
+        // The parameter_events topic is where we'll publish any new/changed/deleted events
+        parameter_events_writer = std::make_shared< dds_topic_writer >( topic, publisher );
+        dds_topic_writer::qos wqos( eprosima::fastdds::dds::RELIABLE_RELIABILITY_QOS );
+        parameter_events_writer->override_qos_from_json( wqos,
+            participant->settings().nested( "device", "ros-parameter-events" ) );
+        parameter_events_writer->run( wqos );
+    }
+
     // Run the LRS device watcher
-    tools::lrs_device_watcher dev_watcher( ctx );
-    dev_watcher.run(
+    auto dev_watcher = std::make_shared< tools::lrs_device_watcher >( ctx );
```

Comment:
Why do we need it as a shared_ptr? It's a local variable

Reply:
True, it's probably a leftover